interface/super/show_groups_popup.php

```diff
@@ -49,7 +49,7 @@
 <ul>
 <?php
 $res = sqlStatement("SELECT distinct(group_name) FROM layout_options WHERE " .
-                    "form_id = '".$_GET['layout_id']."' ORDER BY group_name");
+                    "form_id = ? ORDER BY group_name", array($_GET['layout_id']));
```

Comment:
You should _PROBABLY_ use a prepared statement here to properly fix this.

Reply:
See [this URL](http://bobby-tables.com/php).

Reply:
@robbyoconnor thanks for pointing this out . I was thinking the sqlStatement function was flexible enough for  prepared statements  https://github.com/LibreHealthIO/lh-ehr/blob/master/library/sql.inc#L176

Reply:
> Well, sql.inc could use some modernization...adodb and all that.

@aethelwulffe could you through moe light on this?

Reply:
Oh man.  This stuff involves what is called the Layout Based Forms, or LBF.
It does a few different things.  One, it is the basis of creating forms and fields for demographics type stuff.  It is also used for making somewhat crude custom patient treatment forms, reading the data etc...

This designer can "group" variables it is going to display.  This is part of that tool (I think).  Look at Administration>Layout.

Reply:
I think what robby is looking for here is to go 
$query = "SELECT distinct(group_name) FROM layout_options WHERE " .
                    "form_id = '".$_GET['layout_id']."' ORDER BY group_name");
                    "form_id = ? ORDER BY group_name", array($_GET['layout_id']);";
$res = sqlStatement($query);

Reply:
The above should be viewed as pseudocode.  I just copied and pasted, not actually looked carefully at the syntax in the strings.  It may not work, and if you can't make something like that work, I am not worried about sql injection here so much as there are not really any non enumerated values that get passed to the query itself.

Reply:
Thanks @aethelwulffe

Reply:
This might need something like $res = sqlFetchArray(sqlStatementNoLog($query, array($form_id->id))); or something between the two.  Sorry.  Lazy.

Reply:
Depends on ability to convert to string in sql.inc.